crates/proto/src/lib.rs

```diff
@@ -71,10 +75,50 @@ pub use crate::xfer::secure_dns_handle::SecureDnsHandle;
 #[doc(hidden)]
 pub use crate::xfer::{BufDnsStreamHandle, BufStreamHandle};
 
+#[cfg(feature = "tokio-io")]
+#[doc(hidden)]
+pub mod iocompat {
+    use std::io;
+    use std::pin::Pin;
+    use std::task::{Context, Poll};
+
+    use futures::io::{AsyncRead, AsyncWrite};
+    use tokio::io::{AsyncRead as AsyncRead02, AsyncWrite as AsyncWrite02};
+
+    /// Conversion from `tokio::io::{AsyncRead, AsyncWrite}` to `std::io::{AsyncRead, AsyncWrite}`
+    pub struct Compat02As03<T>(pub T);
+
+    impl<T> Unpin for Compat02As03<T> {}
+    impl<R: AsyncRead02 + Unpin> AsyncRead for Compat02As03<R> {
+        fn poll_read(
+            mut self: Pin<&mut Self>,
+            cx: &mut Context,
+            buf: &mut [u8],
+        ) -> Poll<io::Result<usize>> {
+            Pin::new(&mut self.0).poll_read(cx, buf)
+        }
+    }
+
+    impl<W: AsyncWrite02 + Unpin> AsyncWrite for Compat02As03<W> {
+        fn poll_write(
+            mut self: Pin<&mut Self>,
+            cx: &mut Context<'_>,
+            buf: &[u8],
+        ) -> Poll<io::Result<usize>> {
+            Pin::new(&mut self.0).poll_write(cx, buf)
+        }
+        fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
+            Pin::new(&mut self.0).poll_flush(cx)
+        }
+        fn poll_close(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<io::Result<()>> {
+            unimplemented!()
```

Comment:
This feels dangerous... Also, it will fail CI clippy checks.

Reply:
I am modifying it and have not pushed the new change yet. I will fix other other errors at the same time.